src/main/java/org/bahmni/sms/web/SMSController.java

```diff
@@ -4,26 +4,40 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.bahmni.sms.SMSSender;
+import org.bahmni.sms.impl.DefaultSmsSender;
 import org.bahmni.sms.model.SMSContract;
-import org.bahmni.sms.web.security.OpenMRSAuthenticator;
+import org.bahmni.sms.web.security.TokenValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 @RestController
 @RequestMapping(value = "/notification/sms")
 @AllArgsConstructor
 public class SMSController {
+
+    @Autowired
+    TokenValidator tokenValidator;
+
     private final SMSSender smsSender;
-    private OpenMRSAuthenticator authenticator;
-    private static final Logger logger = LogManager.getLogger(SMSController.class);
+    private static final Logger logger = LogManager.getLogger(DefaultSmsSender.class);
 
     @RequestMapping(method = RequestMethod.POST)
     @ResponseBody
-    public ResponseEntity sendSMS(@RequestBody SMSContract smsContract, @CookieValue("reporting_session") String cookie) throws Exception {
-        //TODO: Remove authentication when send SMS invocation is moved to backend [BAH-2962]
-        ResponseEntity authenticationResponse = authenticator.authenticate(cookie);
-        if (authenticationResponse.getStatusCode().is2xxSuccessful())
+    public ResponseEntity sendSMS(@RequestBody SMSContract smsContract, @RequestHeader(value = HttpHeaders.AUTHORIZATION, required = false) String authorizationHeader) throws Exception {
+        boolean isValidToken = false;
```

Comment:
Shouldn't we leverage Spring Security config which would intercept every request and do authorisation? Adding it in controller may not be a scalable solution in future if we add one more endpoint. @angshu thoughts?

Reply:
Added spring security please review this again